ortools/math_opt/solvers/xpress_solver.cc

```diff
@@ -0,0 +1,693 @@
+                   GetSolutions(model_parameters));
+  for (SolutionProto& solution : solution_and_claims.solutions) {
+    *result.add_solutions() = std::move(solution);
+  }
+  ASSIGN_OR_RETURN(*result.mutable_solve_stats(), GetSolveStats(start));
+  ASSIGN_OR_RETURN(const double best_primal_bound, GetBestPrimalBound());
+  ASSIGN_OR_RETURN(const double best_dual_bound, GetBestDualBound());
+  auto solution_claims = solution_and_claims.solution_claims;
+  ASSIGN_OR_RETURN(*result.mutable_termination(),
+                   ConvertTerminationReason(solution_claims, best_primal_bound,
+                                            best_dual_bound));
+  return result;
+}
+
+absl::StatusOr<double> XpressSolver::GetBestPrimalBound() const {
+  return xpress_->GetDoubleAttr(is_mip_ ? XPRS_MIPOBJVAL : XPRS_LPOBJVAL);
+}
+
+absl::StatusOr<double> XpressSolver::GetBestDualBound() const {
+  // TODO: setting LP primal value as best dual bound. Can this be improved?
+  return xpress_->GetDoubleAttr(XPRS_LPOBJVAL);
+}
+
+absl::StatusOr<XpressSolver::SolutionsAndClaims> XpressSolver::GetSolutions(
+    const ModelSolveParametersProto& model_parameters) {
+  if (is_mip_) {
+    return absl::UnimplementedError("XpressSolver does not handle MIPs yet");
+  } else {
+    return GetLpSolution(model_parameters);
+  }
+}
+
+absl::StatusOr<XpressSolver::SolutionsAndClaims> XpressSolver::GetLpSolution(
+    const ModelSolveParametersProto& model_parameters) {
+  ASSIGN_OR_RETURN(auto primal_solution_and_claim,
+                   GetConvexPrimalSolutionIfAvailable(model_parameters));
+  ASSIGN_OR_RETURN(auto dual_solution_and_claim,
+                   GetConvexDualSolutionIfAvailable(model_parameters));
+  ASSIGN_OR_RETURN(auto basis, GetBasisIfAvailable());
+  const SolutionClaims solution_claims = {
+      .primal_feasible_solution_exists =
+          primal_solution_and_claim.feasible_solution_exists,
+      .dual_feasible_solution_exists =
+          dual_solution_and_claim.feasible_solution_exists};
+
+  if (!primal_solution_and_claim.solution.has_value() &&
+      !dual_solution_and_claim.solution.has_value() && !basis.has_value()) {
+    return SolutionsAndClaims{.solution_claims = solution_claims};
+  }
+  SolutionsAndClaims solution_and_claims{.solution_claims = solution_claims};
+  SolutionProto& solution =
+      solution_and_claims.solutions.emplace_back(SolutionProto());
+  if (primal_solution_and_claim.solution.has_value()) {
+    *solution.mutable_primal_solution() =
+        std::move(*primal_solution_and_claim.solution);
+  }
+  if (dual_solution_and_claim.solution.has_value()) {
+    *solution.mutable_dual_solution() =
+        std::move(*dual_solution_and_claim.solution);
+  }
+  if (basis.has_value()) {
+    *solution.mutable_basis() = std::move(*basis);
+  }
+  return solution_and_claims;
+}
+
+bool XpressSolver::isFeasible() const {
+  if (is_mip_) {
+    return xpress_status_ == XPRS_MIP_OPTIMAL;
```

Comment:
There is another status that indicates "feasible but not proven optimal": `XPRS_MIP_SOLUTION`, see [here](https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/MIPSTATUS.html).